lib/ansible/modules/crypto/get_certificate.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: get_certificate
+author: "John Westcott IV"
+version_added: "2.7"
+short_description: Get a certificate from a host:port
+description:
+    - Makes a secure connection and returns information about the presented certificate
+options:
+    host:
+      description:
+        - The host to get the cert for (IP is fine)
```

Comment:
If you're using an IP, how do you validate the certificate?

Relatedly, do you validate the hostname *at all*?

Reply:
Some testing later: no, the `ssl.get_server_certificate` does not validate the hostname, only the certificate signature. You should probably note this in the docs.

Reply:
We are not doing any kind of validation in this module (other than the signature by passing in ca_file). This is strictly for getting a certificate from a port for inspection. What if we made the module description something like:

Makes a secure connection and returns information about the presented certificate optionally validating the certs signature

Reply:
I think you can add something along the lines of "only validates the certificate in itself, does not validate if the certificate is valid for the supplied host" to the `ca_certs` setting.

Reply:
Sorry for the delay, all set:
```
- ca_certs
        A PEM file containing a list of root certificates; if present, the cert will be validated against these root certs.
        Note that this only validates the certificate is signed by the chain; not that the cert is valid for the host presenting it.
        [Default: (null)]
```

Reply:
@thomwiggers 

> If you're using an IP, how do you validate the certificate?

You can specify subjectAltName to an IP adress and hence have a valid certificate bound to an IP[1] and not the hostname. However I don't think this is widely used. This is if I understood the inital comment, if not please dicard my answer.

[1] https://tools.ietf.org/html/rfc5280#section-4.2.1.6